### PR TITLE
add counter for skip-collision counting

### PR DESCRIPTION
**Pull request by ryanfkeepers**
Introduces a counting bus, and threads it into restore operations so that we can count the number of
collision skips that occur.

---

#### Does this PR need a docs update or release note?

- [x] :no_entry: No

#### Type of change

- [x] :sunflower: Feature

#### Issue(s)

* #3562

#### Test Plan

- [x] :zap: Unit test
- [x] :green_heart: E2E

### CODE REVIEW

File: src/cmd/factory/impl/common.go [hunk 1]
```diff
@@ -63,6 +64,7 @@ func generateAndRestoreItems(
 	dbf dataBuilderFunc,
 	opts control.Options,
 	errs *fault.Bus,
+	cb *count.Bus,
```
Comment (ashmrtn) [inline src/cmd/factory/impl/common.go:67]: nit: use a different variable name. `cb` often represents `callback` which can make skimming confusing

File: src/pkg/count/count.go [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+	}
+
+	if b.parent != nil {
+		return b.parent.Total(k)
+	}
+
+	return b.Get(k)
+}
+
+// Values returns a map of all local values.
```
Comment (ashmrtn) [inline src/pkg/count/count.go:79]: nit: should note this is not a consistent snapshot of values (same with `TotalValues`)

File: src/pkg/count/keys.go [hunk 3]
```diff
@@ -0,0 +1,8 @@
+package count
+
+type key string
+
+const (
+	CollisionSkip key = "collision-skip"
+	Test          key = "just-for-testing"
```
Comment (ashmrtn) [inline src/pkg/count/keys.go:7]: this should be defined only in the test file

**Review by ashmrtn — APPROVED** [review]

File: src/cmd/factory/impl/common.go (hunk 1 above)
Comment (ryanfkeepers) [reply src/cmd/factory/impl/common.go:67]: good thinking, will do

**Pull request merged**